Peer/App/App.cs

```diff
@@ -0,0 +1,61 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using CommandLine;
+using CommandLine.Text;
+using Microsoft.Extensions.DependencyInjection;
+using Peer.App.AppBuilder;
+
+namespace Peer.App;
+
+public class App
+{
+    private readonly CommandLineParser _parser;
+    private readonly List<IVerb> _verbs;
+    private readonly IServiceSetupHandler? _setupHandler;
+
+    public App(CommandLineParser parser, IEnumerable<IVerb> verbs, IServiceSetupHandler? setupHandler = null)
+    {
+        _parser = parser;
+        _verbs = verbs.ToList();
+        _setupHandler = setupHandler;
+
+    }
+
+    public async Task<int> RunAsync(string[] args, CancellationToken token = default)
+    {
+        var result = _parser.Parse(args);
+
+        var verb = _verbs.SelectMany(x => x.Subs.Append(x)).FirstOrDefault(x => x.Type == result.TypeInfo.Current);
+
+        if (verb == null)
+        {
+            await WriteHelpText(null, result);
+            return 1;
+        }
+        var services = new ServiceCollection();
+        _setupHandler?.SetupServices(services);
+
+
+        try
+        {
+            await result.MapResult<object, Task>(
+                x => verb.Handler?.HandleAsync(x, services, token) ?? Task.CompletedTask,
```

Comment:
If verb.Handler is null then the command was invalid right? It would be like doing a command whose only implementation was via subcommands but not specifying a subcommand?

Reply:
Yeah we should be able to assume at this point that it's never null. 

That's exactly it .

config in config (init|show) for ex